quiche/src/recovery/bbr/mod.rs

```diff
@@ -255,13 +255,14 @@ impl State {
 fn bbr_enter_recovery(r: &mut Recovery, now: Instant) {
     r.bbr_state.prior_cwnd = per_ack::bbr_save_cwnd(r);
 
-    r.congestion_window = r.bytes_in_flight +
-        r.bbr_state.newly_acked_bytes.max(r.max_datagram_size);
+    r.congestion_window = r.bytes_in_flight.max(r.max_datagram_size);
```

Comment:
When we see ACKs, the sequence of CC operations called are congestion_event if there are losses and on_packets_acked. From congestion_event we get to know lost_bytes and from on_packets_acked we get to know acked_bytes. The acked_bytes addition to cwnd the above function was doing is https://github.com/cloudflare/quiche/blob/master/quiche/src/recovery/bbr/mod.rs#L300 which is from previous ACK episode. So I have removed that and now cwnd is set to in_flight and we we inflate the cwnd by acked_bytes in https://github.com/cloudflare/quiche/blob/master/quiche/src/recovery/bbr/per_ack.rs#L176 anyways.

Reply:
I see. It looks like linux or quiche does the calculation during ack processing.

To prevent the current issue (using previous acked_bytes), I think we can move the part calling `bbr_enter_recovery()` to `on_packets_acked()` and leave the cwnd calculation to follow the spec better.